pennylane/logging/log_config.toml

```diff
@@ -0,0 +1,102 @@
+
+disable_existing_loggers = false
+version = 1
+
+###############################################################################
+# Bind formatters defined locally and those defined in pennylane.logging
+###############################################################################
+
+[formatters]
+[formatters.qml_default_formatter]
+"()" = "pennylane.logging.formatters.formatter.DefaultFormatter"
+
+[formatters.qml_alt_formatter]
+"()" = "pennylane.logging.formatters.formatter.SimpleFormatter"
+
+[formatters.local_detailed]
+format = "\u001B[38;2;45;145;210m[%(asctime)s][%(levelname)s][<PID %(process)d:%(processName)s>] - %(name)s.%(funcName)s()::\"%(message)s\"\u001B[0m"
+
+[formatters.local_standard]
+format = "[%(asctime)s] - %(name)s - %(levelname)s - %(message)s"
+
+###############################################################################
+# Bind LogRecord filters defined in pennylane.logging module
+###############################################################################
+
+[filters]
+# Filter to show messages from the same local process as the Python script
+[filters.qml_LocalProcessFilter]
+"()" = "pennylane.logging.filter.LocalProcessFilter"
+
+# Filter to show debug level messages only
+[filters.qml_DebugOnlyFilter]
+"()" = "pennylane.logging.filter.DebugOnlyFilter"
+
+###############################################################################
+# Bind handlers defined in the logging and in pennylane.logging modules
+###############################################################################
+
+[handlers]
+[handlers.qml_debug_stream]
+class = "logging.StreamHandler"
+formatter = "qml_default_formatter"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+
+[handlers.qml_debug_stream_alt]
+class = "logging.StreamHandler"
+formatter = "qml_alt_formatter"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+
+[handlers.qml_debug_file]
+class = "logging.handlers.RotatingFileHandler"
+formatter = "local_standard"
+level = "DEBUG"
+filename = "qml_debug.log" # use `/tmp/filename.log` on Linux machines to avoid long-term persistence
+maxBytes = 16777216 # 16MB per file before splitting
+backupCount = 10 # Create 'qml_debug.log.1', ... 'qml_debug.log.backupCount' files and rollover when maxBytes is reached
+delay = true # Avoid file-creation unless used
+
+[handlers.local_filtered_detailed_stdout]
+class = "logging.StreamHandler"
+formatter = "local_standard"
+level = "DEBUG"
+stream = "ext://sys.stdout"
+filters = ["qml_LocalProcessFilter", "qml_DebugOnlyFilter"]
+
+###############################################################################
+# Define logger controls for internal and external packages
+###############################################################################
+
+[loggers]
+
+# Control JAX logging
+[loggers.jax]
+handlers = ["qml_debug_stream",]
+level = "DEBUG"
+propagate = false
+
+# Control logging in the executing Python script
+[loggers.__main__]
```

Comment:
Do we want to influence how the logger in the `"__main__"` file behaves? The main file might be a script that uses a library that depends on PennyLane, no? In this case it would be quite unexpected to suddenly have a different formatting and destination.
On the other hand, this only happens when PennyLane logging is enabled.

Reply:
I just checked, because I remember having the same problem 4 years ago #207 and @josh146 solved this by removing some configurations, but also recommended setting my own logger before importing PennyLane. So I ran
```Python
import logging
from pathlib import Path

logging.basicConfig(
    filename=Path.joinpath(Path(__file__).parent.absolute(), Path('mylogfile.log')),
    filemode='w',
    format='%(asctime)s %(levelname)s:%(message)s',
    level=logging.DEBUG
)
logger = logging.getLogger('foo')

import pennylane as qml

logger.debug('1st foo')
qml.logging.enable_logging()
logger.debug('2nd foo')
```
But my log file still doesn't contain `2nd foo`, just some messages originating from the PL import.
```
2023-08-10 10:10:13,199 DEBUG:matplotlib data path: /Users/frederik.wilde/pennylane/venv3.10-dev/lib/python3.10/site-packages/matplotlib/mpl-data
2023-08-10 10:10:13,202 DEBUG:CONFIGDIR=/Users/frederik.wilde/.matplotlib
2023-08-10 10:10:13,203 DEBUG:interactive is False
2023-08-10 10:10:13,203 DEBUG:platform is darwin
2023-08-10 10:10:13,232 DEBUG:CACHEDIR=/Users/frederik.wilde/.matplotlib
2023-08-10 10:10:13,234 DEBUG:Using fontManager instance from /Users/frederik.wilde/.matplotlib/fontlist-v330.json
2023-08-10 10:10:13,525 DEBUG:Initializing MLIR with module: _site_initialize_0
2023-08-10 10:10:13,525 DEBUG:Registering dialects from initializer <module 'jaxlib.mlir._mlir_libs._site_initialize_0' from '/Users/frederik.wilde/pennylane/venv3.10-dev/lib/python3.10/site-packages/jaxlib/mlir/_mlir_libs/_site_initialize_0.so'>
2023-08-10 10:10:13,586 DEBUG:etils.epath was not found. Using pathlib for file I/O.
2023-08-10 10:10:13,958 DEBUG:1st foo
```

Reply:
I'd only recommend having the pennylane logging setup first --- usually with logging you'd set it at the top of a file, and since we expect user config to adapt the PennyLane config, I'd want to see the pennylane step first, then a suer-define logger. I wouldn't expect a user to ever do:
```
logger.debug('1st foo')
qml.logging.enable_logging()
logger.debug('2nd foo')
```

since logging usually is an end-to-end setup (at least in my experience).

Reply:
Something like:
```
import pennylane as qml
qml.logging.enable_logging()
import logging
from pathlib import Path

logging.basicConfig(
    filename=Path.joinpath(Path(__file__).parent.absolute(), Path('mylogfile.log')),
    filemode='w',
    format='%(asctime)s %(levelname)s:%(message)s',
    level=logging.DEBUG
)
logger = logging.getLogger('foo')

import pennylane as qml

logger.debug('1st foo')
logger.debug('2nd foo')
```

Reply:
Or, what could work is not using 
```
logging.basicConfig(
    filename=Path.joinpath(Path(__file__).parent.absolute(), Path('mylogfile.log')),
    filemode='w',
    format='%(asctime)s %(levelname)s:%(message)s',
    level=logging.DEBUG
)
```
and adding a specific config that doesn't impact the root logging setup. I'll try to create an example and demonstrate.

Reply:
> Something like:

Oh yes, this is definitely be the most reasonable setup! This way I get the desired behaviour. This is probably important to emphasize in the introduction page, so that users are less likely to make the same mistake that I made.

Reply:
This is due to the logging framework in Python itself. The only way to avoid this is to remove all configuration options, which would impact users who simply want to run `qml.logging.enable_logging()` (myself included).

We can likely consider a PEP on how to update Python's logging module to respect multi-config updates, but for now we will need this to be a documentation situation for users to understand the ins and outs of Python's logging infrastructure. As a follow up after this PR is merged, @josh146  suggested we make the dev docs more user facing, and present the users with more guides on @this is how to use it@. I think that approach is the least painful path.